### ISSUE TITLE
Take diagnostic snapshots between gridscan and rotation

### ISSUE THREAD

**Issue by DominicOram** [DiamondLightSource/python-artemis#349]
Between gridscans and data collections we should take 4 snapshots at 0, 90, 180, 270 omega. The path to these is then added to the ispyb entry for the data collection. (see https://ispyb.diamond.ac.uk/dc/visit/cm31105-4/id/9153427).

## Acceptance Criteria
* There is a plan that takes 4 snapshots at the correct locations
* The plan can be performed synchronously or asynchronously

## Notes
* There may be an optimisation where we take the snapshots backwards so we end at 0 for the next scan

**Comment by DominicOram** [comment]
It looks like we're currently only putting 3 snapshots into Hyperion, rather than 4, we should fix that as part of this https://ispyb.diamond.ac.uk/dc/visit/cm37235-2/id/13648572

**Comment by DominicOram** [comment]
Before doing this we will need to do https://github.com/DiamondLightSource/dodal/issues/471. @npat113 is happy with us just doing the crosshairs for now, see https://diamondlightsource.slack.com/archives/C048RD0KFFF/p1714062911126409

**Comment by DominicOram** [comment]
Need to do https://github.com/DiamondLightSource/hyperion/issues/1336 before this

**Comment by dperl-dls** [comment]
Should this be inlined in the rotartion scan plan, or should it be a separate generic plan?

**Comment by DominicOram** [comment]
Yes, let's do a separate plan that we can parameterize cleanly. In which case Large.

**Comment by DominicOram** [comment]
I believe the only part left of this that we don't have code for is:
> There is a way to switch between doing this and doing it in GDA as we currently do
Which is probably easiest done whilst testing on the beamline
